### PR TITLE
[WIP][I-50][R4ML-123] new MLContext API

### PR DESCRIPTION
**Pull request by aloknsingh**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: R4ML/R/ml.model.lm.R [hunk 1]
```diff
@@ -296,6 +294,7 @@ setMethod("stats", signature="r4ml.lm", definition =
 #' @export
 #' @seealso \link{r4ml.lm}
 predict.r4ml.lm <- function(object, data) {
+    browser()
```
Comment (bdwyer2) [inline R4ML/R/ml.model.lm.R]: Forgot to remove

File: R4ML/R/ml.model.lm.R [hunk 2]
```diff
@@ -315,22 +314,26 @@ predict.r4ml.lm <- function(object, data) {
...
-    args <- c(args, dfam = 1)
+    args <- c(args, "$dfam" = 1)
     args <- c(args, B_full = as.r4ml.matrix(coef(object)))
     args <- c(args, "means")
-    args <- c(args, fmt = "csv")
+    args <- c(args, "$fmt" = "csv")
     args <- c(args, dml=file.path(r4ml.env$SYSML_ALGO_ROOT(), r4ml.env$DML_GLM_TEST_SCRIPT))
     
+    r4ml.debug(logSource, "Running GLM predic DML...")
+    r4ml.debugShow(logSource, args)
```
Comment (bdwyer2) [inline R4ML/R/ml.model.lm.R]: I think we should move these debug messages to under `sysml.execute()`

File: R4ML/NAMESPACE [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Generated by roxygen2: do not edit by hand
+# Generated by roxygen2 (4.1.1): do not edit by hand
```
Comment (bdwyer2) [inline R4ML/NAMESPACE]: Can we keep this as is?

File: R4ML/R/zzz.R [hunk 4]
```diff
@@ -239,14 +239,14 @@ sysml.init <- function() {
 
   # static class sysml.RDDUtils
   # since actual class is static, we need this
-  sysml.RDDUtils <- sysml.RDDConverterUtils$new()
-  assign("sysml.RDDUtils", sysml.RDDUtils, .GlobalEnv)
+  #sysml.RDDUtils <- sysml.RDDConverterUtils$new() ##ALOK TODO rm
+  #assign("sysml.RDDUtils", sysml.RDDUtils, .GlobalEnv) ## ALOK TODO rm
```
Comment (bdwyer2) [inline R4ML/R/zzz.R]: Did you mean to remove these?

File: R4ML/inst/sysml/scripts/algorithms/GLM-predict.dml [hunk 5]
```diff
@@ -86,14 +86,21 @@
...
 fileM = ifdef ($M, " ");
 fileY = " ";
 fileO = ifdef ($O, " ");
 fmtM  = ifdef ($fmt, "text");
 cmd_scoring_only = ifdef($scoring_only, "yes");
 
+print("ALOK")
+print(cmd_scoring_only)
+
+stop("test ")
```
Comment (bdwyer2) [inline R4ML/inst/sysml/scripts/algorithms/GLM-predict.dml]: Did you mean to keep this?

**Comment by bdwyer2** [comment]
Failed tests:
```
OK: 679 SKIPPED: 9 FAILED: 53
1. Error: r4ml.als (@test-ml.model.als.R#27) 
2. Error: r4ml.als_regularization (@test-ml.model.als.R#43) 
3. Error: r4ml.als_weighted_regularization (@test-ml.model.als.R#59) 
4. Error: r4ml.als_altered_rank (@test-ml.model.als.R#75) 
5. Error: r4ml.als_all_parameters (@test-ml.model.als.R#91) 
6. Error: r4ml.coxph model (@test-ml.model.cox.R#74) 
7. Error: r4ml.glm (@test-ml.model.glm.R#29) 
8. Error: predict.r4ml.glm (@test-ml.model.glm.R#59) 
9. Error: r4ml.glm accuracy (@test-ml.model.glm.R#89) 
1. ...
```

**Review by bdwyer2 — DISMISSED** [review]
Should we also include a newer version of SystemML with this PR?

File: R4ML/R/sysml.bridge.R [hunk 6]
```diff
@@ -50,6 +50,194 @@ sysml.MatrixCharacteristics <- setRefClass("sysml.MatrixCharacteristics",
 
 )
 
+#' @export
```
Comment (bdwyer2) [inline R4ML/R/sysml.bridge.R:53]: I think we should add some documentation

File: R4ML/R/sysml.bridge.R [hunk 7]
```diff
@@ -50,6 +50,194 @@ sysml.MatrixCharacteristics <- setRefClass("sysml.MatrixCharacteristics",
...
+      env <<- new.env()
+      env$jref <<- SparkR:::newJObject("org.apache.sysml.api.mlcontext.ScriptFactory")
+    },
+    finalize = function() {
+      SparkR:::cleanup.jobj(env$jref)
+      
+    },
+    
+    dmlFromString = function(dml) {
+      script_jref <- SparkR:::callJMethod(.self$env$jref, "dmlFromString", dml)
```
Comment (bdwyer2) [inline R4ML/R/sysml.bridge.R]: We should use the public version of this function

File: R4ML/R/sysml.bridge.R [hunk 8]
```diff
@@ -50,6 +50,194 @@ sysml.MatrixCharacteristics <- setRefClass("sysml.MatrixCharacteristics",
...
+      return(sysml.Script$new(jref = script_jref))
+    },
+    
+    dmlFromLocalFile = function(dmlFile) {
+      script_jref <- SparkR:::callJMethod(.self$env$jref, "dmlFromFile", dmlFile)
```
Comment (bdwyer2) [inline R4ML/R/sysml.bridge.R]: We should use the public version of this function

File: R4ML/R/sysml.bridge.R (hunk 6 above)
Comment (aloknsingh) [reply R4ML/R/sysml.bridge.R:53]: @bdwyer2 see the jira it says WIP, and it especifically says that it is not yet ready

Comment (bdwyer2) [reply R4ML/R/sysml.bridge.R:53]: Sorry. The PR made no mention of it being a work in progress.

File: R4ML/R/ml.model.lm.R (hunk 2 above)
Comment (aloknsingh) [reply R4ML/R/ml.model.lm.R]: ok it sounds good that we mv r4ml.debugShow(logSource, args) to execute

**Pull request closed without merging**